### PR TITLE
refactor(streaming): stream agg only read and write value column

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As we have changed streaming table's read and write interfaces to handle value indices (default is full columns), this PR changes `stream agg` to only read and write one value column. Previous PR have refactor hash_join's degree table.

After the refactor of `stream agg` is done by @st1page , the `result_table` is composed of `group_key` and all `agg_call`'s values, so the `value_indices` of this table should be `[group_key.len() .. columns().len()-1]`.
## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
https://github.com/risingwavelabs/risingwave/issues/5101

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/logical_agg.rs [hunk 1]
```diff
@@ -377,11 +379,13 @@ impl LogicalAgg {
                 column_mapping.push(idx);
             }
             internal_table_catalog_builder.add_column(&out_fields[value_key]);
+            let value_indices = vec![internal_table_catalog_builder.get_columns().len() - 1];
```
Comment (wcy-fdu) [inline src/frontend/src/optimizer/plan_node/logical_agg.rs:382]: Not sure whether value column of `agg` is always the last column.

**Comment by st1page** [comment]
what do you mean about `local agg`?

**Comment by st1page** [comment]
this PR changes the definition of `column_mapping`. I think we need more discussion later.

File: src/stream/src/executor/aggregation/mod.rs [hunk 2]
```diff
@@ -283,13 +283,10 @@ pub async fn generate_managed_agg_state<S: StateStore>(
...
     extreme_cache_size: usize,
 ) -> StreamExecutorResult<AggState<S>> {
-    let group_key_len = group_key.as_ref().map_or(0, |row| row.size());
-
     let prev_result: Option<Row> = result_table
         .get_row(group_key.as_ref().unwrap_or_else(Row::empty))
         .await?;
-    let prev_outputs: Option<Vec<_>> =
-        prev_result.map(|row| row.0.into_iter().skip(group_key_len).collect());
+    let prev_outputs: Option<Vec<_>> = prev_result.map(|row| row.0.into_iter().collect());
```
Comment (BugenZhao) [inline src/stream/src/executor/aggregation/mod.rs:289]: No need to `into_iter().collect()` here?

**Comment by wcy-fdu** [comment]
Some planner_tests fail
```
   stream_dist_agg_test
   nexmark_test
   distribution_derive_test
   tpch_test
```

**Comment by wcy-fdu** [comment]
e2e tests do not seem to cover this change, related to https://github.com/risingwavelabs/risingwave/issues/5080

**Comment by BugenZhao** [comment]
> e2e tests do not seem to cover this change, related to https://github.com/risingwavelabs/risingwave/issues/5080

The newly introduced scaling test will cover this. :)

File: src/frontend/src/optimizer/plan_node/logical_agg.rs [hunk 3]
```diff
@@ -367,6 +367,9 @@ impl LogicalAgg {
         if let Some(tb_vnode_idx) = vnode_col_idx.and_then(|idx| mapping.try_map(idx)) {
             internal_table_catalog_builder.set_vnode_col_idx(tb_vnode_idx);
         }
+        internal_table_catalog_builder.set_value_indices(
+            (self.group_key().len()..internal_table_catalog_builder.columns().len()).collect_vec(),
+        );
```
Comment (richardchien) [inline src/frontend/src/optimizer/plan_node/logical_agg.rs:372]: Will the following be more concise?

```rust
internal_table_catalog_builder.set_value_indices((self.group_key().len()..out_fields.len()).collect());
```

File: src/stream/src/executor/aggregation/mod.rs [hunk 4]
```diff
@@ -283,13 +283,10 @@ pub async fn generate_managed_agg_state<S: StateStore>(
     pk_indices: &PkIndices,
     extreme_cache_size: usize,
 ) -> StreamExecutorResult<AggState<S>> {
-    let group_key_len = group_key.as_ref().map_or(0, |row| row.size());
-
     let prev_result: Option<Row> = result_table
```
Comment (richardchien) [inline src/stream/src/executor/aggregation/mod.rs:286]: May be better to add comment here to explain the skipped group key prefix. :)

**Review by richardchien — APPROVED** [review]
LGTM!

**Comment by richardchien** [comment]
`MaterializedInputState`s need this too, I may do that later, together with some other refactoring.

**Pull request merged**